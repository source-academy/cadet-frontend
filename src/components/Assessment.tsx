import { Button, Dialog, NonIdealState, Spinner } from '@blueprintjs/core'
import { IconNames } from '@blueprintjs/icons'
import * as React from 'react'

import Workspace from '../containers/workspace'
import { SideContentTab } from './workspace/side-content'

export type AssessmentInfo = {
  longSummary: string
  dueDate: string
  studentBriefed: boolean
}

export type StateProps = {
  missionId: number
  assessmentInfo?: AssessmentInfo
}

export type DispatchProps = {
  handleAssessmentInfoFetch: (missionId: number) => void
}

export type AssessmentProps = StateProps & DispatchProps

class Assessment extends React.Component<AssessmentProps, { showOverlay: boolean }> {
  public state = { showOverlay: true }

  public render() {
    if (this.props.assessmentInfo === undefined) {
      this.props.handleAssessmentInfoFetch(this.props.missionId)
      return <NonIdealState description="Getting mission ready..." visual={<Spinner />} />
    }
    const briefing = <> ({this.props.assessmentInfo.longSummary}) </>
    const overlay = (
      <Dialog
        className="mission-briefing"
        isOpen={this.state.showOverlay && !this.props.assessmentInfo.studentBriefed}
      >
        <>
          {briefing}
          <Button
            // tslint:disable-next-line jsx-no-lambda
            onClick={() => this.setState({ showOverlay: false })}
            text="Primary"
          />
        </>
      </Dialog>
    )
    const briefingTab: SideContentTab = {
      label: 'Briefing',
      icon: IconNames.BRIEFCASE,
      body: briefing
    }
    return (
      <>
        {overlay}
        <Workspace sideContentTabs={[briefingTab]} />
      </>
    )
  }
}

export default Assessment
