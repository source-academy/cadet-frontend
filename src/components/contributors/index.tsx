import * as React from 'react';
import ContributorsDetails from './ContributorsDetails';
import ContributorsList from './ContributorsList';

export class Contributors extends React.Component {
  public render() {
    return (
      <div>
        <ContributorsDetails />
        <ContributorsList />
      </div>
    );
  }
}

export default Contributors;
