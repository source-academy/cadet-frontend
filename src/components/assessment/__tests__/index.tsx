import { mount } from 'enzyme';
import * as React from 'react';
import { MemoryRouter } from 'react-router';

import Assessment, { IAssessmentProps } from '../';
import { mockAssessmentOverviews } from '../../../mocks/assessmentAPI';
import { mockRouterProps } from '../../../mocks/components';
import { AssessmentCategories } from '../assessmentShape';

const defaultProps: IAssessmentProps = {
  assessmentCategory: AssessmentCategories.Mission,
  assessmentOverviews: undefined,
  handleAssessmentOverviewFetch: () => {},
  handleSubmitAssessment: (id: number) => {},
  isStudent: false,
  ...mockRouterProps('/academy/missions', {})
};

const mockUndefinedAssessment: IAssessmentProps = {
  ...defaultProps,
  assessmentOverviews: undefined
};

const mockEmptyAssessment: IAssessmentProps = {
  ...defaultProps,
  assessmentOverviews: []
};

const mockPresentAssessment: IAssessmentProps = {
  ...defaultProps,
  assessmentOverviews: mockAssessmentOverviews
};

const mockPresentAssessmentForStudent: IAssessmentProps = {
  ...defaultProps,
  assessmentOverviews: mockAssessmentOverviews,
  isStudent: true
};

test('Assessment page "loading" content renders correctly', () => {
  const app = (
    <MemoryRouter initialEntries={['/unknown']}>
      <Assessment {...mockUndefinedAssessment} />
    </MemoryRouter>
  );
  const tree = mount(app);
  expect(tree.debug()).toMatchSnapshot();
});

test('Assessment page with 0 missions renders correctly', () => {
  const app = (
    <MemoryRouter initialEntries={['/unknown']}>
      <Assessment {...mockEmptyAssessment} />
    </MemoryRouter>
  );
  const tree = mount(app);
  expect(tree.debug()).toMatchSnapshot();
});

test('Assessment page with multiple loaded missions renders correctly', () => {
  const app = (
    <MemoryRouter initialEntries={['/unknown']}>
      <Assessment {...mockPresentAssessment} />
    </MemoryRouter>
  );
  const tree = mount(app);
  expect(tree.debug()).toMatchSnapshot();
});

test('Assessment page does not show attempt Button for upcoming assessments for student user', () => {
  const app = (
    <MemoryRouter initialEntries={['/unknown']}>
      <Assessment {...mockPresentAssessmentForStudent} />
    </MemoryRouter>
  );
  const tree = mount(app);
  expect(tree.debug()).toMatchSnapshot();
});
