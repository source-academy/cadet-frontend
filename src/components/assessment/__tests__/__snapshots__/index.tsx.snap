// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assessment page "loading" content renders correctly 1`] = `
"<MemoryRouter initialEntries={{...}}>
  <Router history={{...}}>
    <Assessment assessmentCategory=\\"Mission\\" assessmentOverviews={[undefined]} handleAssessmentOverviewFetch={[Function: handleAssessmentOverviewFetch]} match={{...}} location={{...}} history={{...}} staticContext={{...}}>
      <div className=\\"Assessment\\">
        <ContentDisplay display={{...}} loadContentDispatch={[Function: handleAssessmentOverviewFetch]}>
          <div className=\\"ContentDisplay row center-xs\\">
            <div className=\\"col-xs-10 contentdisplay-content-parent\\">
              <Blueprint2.Card className=\\"contentdisplay-content\\" elevation={3} interactive={false}>
                <div className=\\"pt-card pt-elevation-3 contentdisplay-content\\">
                  <NonIdealState description=\\"Fetching assessment...\\" visual={{...}}>
                    <div className=\\"pt-non-ideal-state\\">
                      <div className=\\"pt-non-ideal-state-visual\\">
                        <Blueprint2.Spinner>
                          <div className=\\"pt-spinner\\">
                            <div className=\\"pt-spinner-svg-container\\">
                              <svg viewBox=\\"0 0 100 100\\">
                                <path className=\\"pt-spinner-track\\" d=\\"M 50,50 m 0,-44.5 a 44.5,44.5 0 1 1 0,89 a 44.5,44.5 0 1 1 0,-89\\" />
                                <path className=\\"pt-spinner-head\\" d=\\"M 50,50 m 0,-44.5 a 44.5,44.5 0 1 1 0,89 a 44.5,44.5 0 1 1 0,-89\\" pathLength={280} style={{...}} />
                              </svg>
                            </div>
                          </div>
                        </Blueprint2.Spinner>
                      </div>
                      <div className=\\"pt-non-ideal-state-description\\">
                        Fetching assessment...
                      </div>
                    </div>
                  </NonIdealState>
                </div>
              </Blueprint2.Card>
            </div>
          </div>
        </ContentDisplay>
      </div>
    </Assessment>
  </Router>
</MemoryRouter>"
`;

exports[`Assessment page with 0 missions renders correctly 1`] = `
"<MemoryRouter initialEntries={{...}}>
  <Router history={{...}}>
    <Assessment assessmentCategory=\\"Mission\\" assessmentOverviews={{...}} handleAssessmentOverviewFetch={[Function: handleAssessmentOverviewFetch]} match={{...}} location={{...}} history={{...}} staticContext={{...}}>
      <div className=\\"Assessment\\">
        <ContentDisplay display={{...}} loadContentDispatch={[Function: handleAssessmentOverviewFetch]}>
          <div className=\\"ContentDisplay row center-xs\\">
            <div className=\\"col-xs-10 contentdisplay-content-parent\\">
              <Blueprint2.Card className=\\"contentdisplay-content\\" elevation={3} interactive={false}>
                <div className=\\"pt-card pt-elevation-3 contentdisplay-content\\">
                  <NonIdealState title=\\"There are no assessments.\\" visual=\\"flame\\">
                    <div className=\\"pt-non-ideal-state\\">
                      <div className=\\"pt-non-ideal-state-visual pt-non-ideal-state-icon\\">
                        <Blueprint2.Icon icon=\\"flame\\" iconSize={60}>
                          <svg className=\\"pt-icon\\" style={{...}} data-icon=\\"flame\\" width={60} height={60} viewBox=\\"0 0 20 20\\">
                            <title>
                              flame
                            </title>
                            <path d=\\"M11.622 0c0 1.71.49 3.077 1.472 4.103C16.364 6.496 18 9.23 18 12.308c0 3.418-1.962 5.983-5.887 7.692 2.887-3 2.453-4.23-.49-8C8.5 13.5 9 14.5 9.5 16.5c-1.048 0-2 0-2.5-.5 0 .684 1.197 2.5 1.952 4-3.924-1.026-8.123-7.18-6.651-7.692.981-.342 2.126-.171 3.434.513C4.1 6.667 6.062 2.393 11.622 0z\\" fillRule=\\"evenodd\\" />
                          </svg>
                        </Blueprint2.Icon>
                      </div>
                      <h4 className=\\"pt-non-ideal-state-title\\">
                        There are no assessments.
                      </h4>
                    </div>
                  </NonIdealState>
                </div>
              </Blueprint2.Card>
            </div>
          </div>
        </ContentDisplay>
      </div>
    </Assessment>
  </Router>
</MemoryRouter>"
`;

exports[`Assessment page with multiple loaded missions renders correctly 1`] = `
"<MemoryRouter initialEntries={{...}}>
  <Router history={{...}}>
    <Assessment assessmentCategory=\\"Mission\\" assessmentOverviews={{...}} handleAssessmentOverviewFetch={[Function: handleAssessmentOverviewFetch]} match={{...}} location={{...}} history={{...}} staticContext={{...}}>
      <div className=\\"Assessment\\">
        <ContentDisplay display={{...}} loadContentDispatch={[Function: handleAssessmentOverviewFetch]}>
          <div className=\\"ContentDisplay row center-xs\\">
            <div className=\\"col-xs-10 contentdisplay-content-parent\\">
              <Blueprint2.Card className=\\"contentdisplay-content\\" elevation={3} interactive={false}>
                <div className=\\"pt-card pt-elevation-3 contentdisplay-content\\">
                  <Blueprint2.Button onClick={[Function]} intent=\\"none\\" minimal={true} className=\\"collapse-button\\" icon=\\"caret-down\\">
                    <button type=\\"button\\" onClick={[Function]} className=\\"pt-button pt-minimal collapse-button\\" disabled={[undefined]} onKeyDown={[Function]} onKeyUp={[Function]}>
                      <Blueprint2.Icon icon=\\"caret-down\\">
                        <svg className=\\"pt-icon\\" style={{...}} data-icon=\\"caret-down\\" width={16} height={16} viewBox=\\"0 0 16 16\\">
                          <title>
                            caret-down
                          </title>
                          <path d=\\"M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 0 0-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z\\" fillRule=\\"evenodd\\" />
                        </svg>
                      </Blueprint2.Icon>
                      <span className=\\"pt-button-text\\">
                        Due soon
                      </span>
                      <Blueprint2.Icon icon={[undefined]} />
                    </button>
                  </Blueprint2.Button>
                  <Blueprint2.Collapse isOpen={true} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                    <div className=\\"pt-collapse\\" style={{...}}>
                      <div className=\\"pt-collapse-body\\" style={{...}} aria-hidden={false}>
                        <div>
                          <Blueprint2.Card className=\\"row listing\\" elevation={1} interactive={false}>
                            <div className=\\"pt-card pt-elevation-1 row listing\\">
                              <div className=\\"col-xs-3 listing-picture\\">
                                <img src=\\"www.imgur.com\\" />
                              </div>
                              <div className=\\"col-xs-9 listing-text\\">
                                <div className=\\"row listing-title\\">
                                  <Text ellipsize={true}>
                                    <div className=\\"pt-text-overflow-ellipsis\\" title={[undefined]}>
                                      <h4>
                                        An Odessey to Runes
                                      </h4>
                                    </div>
                                  </Text>
                                  <Blueprint2.Popover content={{...}} defaultIsOpen={false} disabled={false} hasBackdrop={false} hoverCloseDelay={300} hoverOpenDelay={150} inheritDarkTheme={true} interactionKind=\\"click\\" minimal={false} modifiers={{...}} openOnTargetFocus={true} position=\\"auto\\" rootElementTag=\\"span\\" targetElementTag=\\"div\\" transitionDuration={300} usePortal={true}>
                                    <Manager tag=\\"span\\" className=\\"pt-popover-wrapper\\">
                                      <span className=\\"pt-popover-wrapper\\">
                                        <Target onClick={[Function]} className=\\"pt-popover-target\\" component=\\"div\\" innerRef={[Function: target]}>
                                          <div onClick={[Function]} className=\\"pt-popover-target\\">
                                            <Blueprint2.Button icon=\\"menu\\" minimal={true} className=\\"\\" disabled={[undefined]} tabIndex={[undefined]}>
                                              <button type=\\"button\\" className=\\"pt-button pt-minimal\\" disabled={[undefined]} tabIndex={[undefined]} onClick={[undefined]} onKeyDown={[Function]} onKeyUp={[Function]}>
                                                <Blueprint2.Icon icon=\\"menu\\">
                                                  <svg className=\\"pt-icon\\" style={{...}} data-icon=\\"menu\\" width={16} height={16} viewBox=\\"0 0 16 16\\">
                                                    <title>
                                                      menu
                                                    </title>
                                                    <path d=\\"M1 4h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 8H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-5H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z\\" fillRule=\\"evenodd\\" />
                                                  </svg>
                                                </Blueprint2.Icon>
                                                <Blueprint2.Icon icon={[undefined]} />
                                              </button>
                                            </Blueprint2.Button>
                                          </div>
                                        </Target>
                                        <Blueprint2.Overlay autoFocus={true} backdropClassName=\\"pt-popover-backdrop\\" backdropProps={{...}} canEscapeKeyClose={true} canOutsideClickClose={true} className={[undefined]} didOpen={[Function]} enforceFocus={true} hasBackdrop={false} usePortal={true} isOpen={false} onClose={[Function]} transitionDuration={300} transitionName=\\"pt-popover\\" lazy={true} />
                                      </span>
                                    </Manager>
                                  </Blueprint2.Popover>
                                </div>
                                <div className=\\"row listing-order\\">
                                  <h6>
                                    Max Grade: 3000
                                  </h6>
                                </div>
                                <div className=\\"row listing-description\\">
                                  <Text className=\\"col-xs-12\\" ellipsize={true}>
                                    <div className=\\"pt-text-overflow-ellipsis col-xs-12\\" title={[undefined]}>
                                      Once upon a time, Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin nec vulputate sapien. Fusce vel lacus fermentum, efficitur ipsum.
                                    </div>
                                  </Text>
                                </div>
                                <div className=\\"listing-controls\\">
                                  <Text className=\\"listing-due-date\\">
                                    <div className=\\"listing-due-date\\" title={[undefined]}>
                                      <Blueprint2.Icon className=\\"listing-due-icon\\" iconSize={12} icon=\\"time\\">
                                        <svg className=\\"pt-icon listing-due-icon\\" style={{...}} data-icon=\\"time\\" width={12} height={12} viewBox=\\"0 0 16 16\\">
                                          <title>
                                            time
                                          </title>
                                          <path d=\\"M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm1-6.41V4c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .28.11.53.29.71l2 2a1.003 1.003 0 0 0 1.42-1.42L9 7.59z\\" fillRule=\\"evenodd\\" />
                                        </svg>
                                      </Blueprint2.Icon>
                                      Due: 18/5, 13:24
                                    </div>
                                  </Text>
                                  <NavLink to=\\"/academy/missions/0/0\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
                                    <Route path=\\"/academy/missions/0/0\\" exact={[undefined]} strict={[undefined]} location={[undefined]}>
                                      <Link to=\\"/academy/missions/0/0\\" className={[undefined]} style={[undefined]} aria-current={false} replace={false}>
                                        <a className={[undefined]} style={[undefined]} aria-current={false} onClick={[Function]} href=\\"/academy/missions/0/0\\">
                                          <Blueprint2.Button className=\\"listing-skip-button\\" minimal={true} intent=\\"primary\\" icon=\\"flame\\">
                                            <button type=\\"button\\" className=\\"pt-button pt-minimal pt-intent-primary listing-skip-button\\" disabled={[undefined]} onClick={[undefined]} onKeyDown={[Function]} onKeyUp={[Function]}>
                                              <Blueprint2.Icon icon=\\"flame\\">
                                                <svg className=\\"pt-icon\\" style={{...}} data-icon=\\"flame\\" width={16} height={16} viewBox=\\"0 0 16 16\\">
                                                  <title>
                                                    flame
                                                  </title>
                                                  <path d=\\"M9.217 0c0 1.368.368 2.462 1.104 3.282C12.774 5.197 14 7.385 14 9.846c0 2.735-1.472 4.786-4.415 6.154 2.165-2.4 1.84-3.385-.368-6.4-2.342 1.2-1.967 2-1.592 3.6-.786 0-1.5 0-1.875-.4 0 .547.898 2 1.464 3.2-2.943-.82-6.092-5.744-4.988-6.154.736-.273 1.594-.137 2.575.41C3.575 5.333 5.047 1.915 9.217 0z\\" fillRule=\\"evenodd\\" />
                                                </svg>
                                              </Blueprint2.Icon>
                                              <span className=\\"pt-button-text\\">
                                                Skip Story &amp; Attempt
                                              </span>
                                              <Blueprint2.Icon icon={[undefined]} />
                                            </button>
                                          </Blueprint2.Button>
                                        </a>
                                      </Link>
                                    </Route>
                                  </NavLink>
                                </div>
                              </div>
                            </div>
                          </Blueprint2.Card>
                        </div>
                        <div>
                          <Blueprint2.Card className=\\"row listing\\" elevation={1} interactive={false}>
                            <div className=\\"pt-card pt-elevation-1 row listing\\">
                              <div className=\\"col-xs-3 listing-picture\\">
                                <img src=\\"www.imgur.com\\" />
                              </div>
                              <div className=\\"col-xs-9 listing-text\\">
                                <div className=\\"row listing-title\\">
                                  <Text ellipsize={true}>
                                    <div className=\\"pt-text-overflow-ellipsis\\" title={[undefined]}>
                                      <h4>
                                        The Secret to Streams
                                      </h4>
                                    </div>
                                  </Text>
                                  <Blueprint2.Popover content={{...}} defaultIsOpen={false} disabled={false} hasBackdrop={false} hoverCloseDelay={300} hoverOpenDelay={150} inheritDarkTheme={true} interactionKind=\\"click\\" minimal={false} modifiers={{...}} openOnTargetFocus={true} position=\\"auto\\" rootElementTag=\\"span\\" targetElementTag=\\"div\\" transitionDuration={300} usePortal={true}>
                                    <Manager tag=\\"span\\" className=\\"pt-popover-wrapper\\">
                                      <span className=\\"pt-popover-wrapper\\">
                                        <Target onClick={[Function]} className=\\"pt-popover-target\\" component=\\"div\\" innerRef={[Function: target]}>
                                          <div onClick={[Function]} className=\\"pt-popover-target\\">
                                            <Blueprint2.Button icon=\\"menu\\" minimal={true} className=\\"\\" disabled={[undefined]} tabIndex={[undefined]}>
                                              <button type=\\"button\\" className=\\"pt-button pt-minimal\\" disabled={[undefined]} tabIndex={[undefined]} onClick={[undefined]} onKeyDown={[Function]} onKeyUp={[Function]}>
                                                <Blueprint2.Icon icon=\\"menu\\">
                                                  <svg className=\\"pt-icon\\" style={{...}} data-icon=\\"menu\\" width={16} height={16} viewBox=\\"0 0 16 16\\">
                                                    <title>
                                                      menu
                                                    </title>
                                                    <path d=\\"M1 4h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 8H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-5H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z\\" fillRule=\\"evenodd\\" />
                                                  </svg>
                                                </Blueprint2.Icon>
                                                <Blueprint2.Icon icon={[undefined]} />
                                              </button>
                                            </Blueprint2.Button>
                                          </div>
                                        </Target>
                                        <Blueprint2.Overlay autoFocus={true} backdropClassName=\\"pt-popover-backdrop\\" backdropProps={{...}} canEscapeKeyClose={true} canOutsideClickClose={true} className={[undefined]} didOpen={[Function]} enforceFocus={true} hasBackdrop={false} usePortal={true} isOpen={false} onClose={[Function]} transitionDuration={300} transitionName=\\"pt-popover\\" lazy={true} />
                                      </span>
                                    </Manager>
                                  </Blueprint2.Popover>
                                </div>
                                <div className=\\"row listing-order\\">
                                  <h6>
                                    Max Grade: 3000
                                  </h6>
                                </div>
                                <div className=\\"row listing-description\\">
                                  <Text className=\\"col-xs-12\\" ellipsize={true}>
                                    <div className=\\"pt-text-overflow-ellipsis col-xs-12\\" title={[undefined]}>
                                      Once upon a time, Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin nec vulputate sapien. Fusce vel lacus fermentum, efficitur ipsum.
                                    </div>
                                  </Text>
                                </div>
                                <div className=\\"listing-controls\\">
                                  <Text className=\\"listing-due-date\\">
                                    <div className=\\"listing-due-date\\" title={[undefined]}>
                                      <Blueprint2.Icon className=\\"listing-due-icon\\" iconSize={12} icon=\\"time\\">
                                        <svg className=\\"pt-icon listing-due-icon\\" style={{...}} data-icon=\\"time\\" width={12} height={12} viewBox=\\"0 0 16 16\\">
                                          <title>
                                            time
                                          </title>
                                          <path d=\\"M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm1-6.41V4c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .28.11.53.29.71l2 2a1.003 1.003 0 0 0 1.42-1.42L9 7.59z\\" fillRule=\\"evenodd\\" />
                                        </svg>
                                      </Blueprint2.Icon>
                                      Due: 18/5, 13:24
                                    </div>
                                  </Text>
                                  <NavLink to=\\"/academy/missions/1/0\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
                                    <Route path=\\"/academy/missions/1/0\\" exact={[undefined]} strict={[undefined]} location={[undefined]}>
                                      <Link to=\\"/academy/missions/1/0\\" className={[undefined]} style={[undefined]} aria-current={false} replace={false}>
                                        <a className={[undefined]} style={[undefined]} aria-current={false} onClick={[Function]} href=\\"/academy/missions/1/0\\">
                                          <Blueprint2.Button className=\\"listing-skip-button\\" minimal={true} intent=\\"primary\\" icon=\\"flame\\">
                                            <button type=\\"button\\" className=\\"pt-button pt-minimal pt-intent-primary listing-skip-button\\" disabled={[undefined]} onClick={[undefined]} onKeyDown={[Function]} onKeyUp={[Function]}>
                                              <Blueprint2.Icon icon=\\"flame\\">
                                                <svg className=\\"pt-icon\\" style={{...}} data-icon=\\"flame\\" width={16} height={16} viewBox=\\"0 0 16 16\\">
                                                  <title>
                                                    flame
                                                  </title>
                                                  <path d=\\"M9.217 0c0 1.368.368 2.462 1.104 3.282C12.774 5.197 14 7.385 14 9.846c0 2.735-1.472 4.786-4.415 6.154 2.165-2.4 1.84-3.385-.368-6.4-2.342 1.2-1.967 2-1.592 3.6-.786 0-1.5 0-1.875-.4 0 .547.898 2 1.464 3.2-2.943-.82-6.092-5.744-4.988-6.154.736-.273 1.594-.137 2.575.41C3.575 5.333 5.047 1.915 9.217 0z\\" fillRule=\\"evenodd\\" />
                                                </svg>
                                              </Blueprint2.Icon>
                                              <span className=\\"pt-button-text\\">
                                                Skip Story &amp; Attempt
                                              </span>
                                              <Blueprint2.Icon icon={[undefined]} />
                                            </button>
                                          </Blueprint2.Button>
                                        </a>
                                      </Link>
                                    </Route>
                                  </NavLink>
                                </div>
                              </div>
                            </div>
                          </Blueprint2.Card>
                        </div>
                        <div>
                          <Blueprint2.Card className=\\"row listing\\" elevation={1} interactive={false}>
                            <div className=\\"pt-card pt-elevation-1 row listing\\">
                              <div className=\\"col-xs-3 listing-picture\\">
                                <img src=\\"www.imgur.com\\" />
                              </div>
                              <div className=\\"col-xs-9 listing-text\\">
                                <div className=\\"row listing-title\\">
                                  <Text ellipsize={true}>
                                    <div className=\\"pt-text-overflow-ellipsis\\" title={[undefined]}>
                                      <h4>
                                        A sample Sidequest
                                      </h4>
                                    </div>
                                  </Text>
                                  <Blueprint2.Popover content={{...}} defaultIsOpen={false} disabled={false} hasBackdrop={false} hoverCloseDelay={300} hoverOpenDelay={150} inheritDarkTheme={true} interactionKind=\\"click\\" minimal={false} modifiers={{...}} openOnTargetFocus={true} position=\\"auto\\" rootElementTag=\\"span\\" targetElementTag=\\"div\\" transitionDuration={300} usePortal={true}>
                                    <Manager tag=\\"span\\" className=\\"pt-popover-wrapper\\">
                                      <span className=\\"pt-popover-wrapper\\">
                                        <Target onClick={[Function]} className=\\"pt-popover-target\\" component=\\"div\\" innerRef={[Function: target]}>
                                          <div onClick={[Function]} className=\\"pt-popover-target\\">
                                            <Blueprint2.Button icon=\\"menu\\" minimal={true} className=\\"\\" disabled={[undefined]} tabIndex={[undefined]}>
                                              <button type=\\"button\\" className=\\"pt-button pt-minimal\\" disabled={[undefined]} tabIndex={[undefined]} onClick={[undefined]} onKeyDown={[Function]} onKeyUp={[Function]}>
                                                <Blueprint2.Icon icon=\\"menu\\">
                                                  <svg className=\\"pt-icon\\" style={{...}} data-icon=\\"menu\\" width={16} height={16} viewBox=\\"0 0 16 16\\">
                                                    <title>
                                                      menu
                                                    </title>
                                                    <path d=\\"M1 4h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 8H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-5H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z\\" fillRule=\\"evenodd\\" />
                                                  </svg>
                                                </Blueprint2.Icon>
                                                <Blueprint2.Icon icon={[undefined]} />
                                              </button>
                                            </Blueprint2.Button>
                                          </div>
                                        </Target>
                                        <Blueprint2.Overlay autoFocus={true} backdropClassName=\\"pt-popover-backdrop\\" backdropProps={{...}} canEscapeKeyClose={true} canOutsideClickClose={true} className={[undefined]} didOpen={[Function]} enforceFocus={true} hasBackdrop={false} usePortal={true} isOpen={false} onClose={[Function]} transitionDuration={300} transitionName=\\"pt-popover\\" lazy={true} />
                                      </span>
                                    </Manager>
                                  </Blueprint2.Popover>
                                </div>
                                <div className=\\"row listing-order\\">
                                  <h6>
                                    Max Grade: 3000
                                  </h6>
                                </div>
                                <div className=\\"row listing-description\\">
                                  <Text className=\\"col-xs-12\\" ellipsize={true}>
                                    <div className=\\"pt-text-overflow-ellipsis col-xs-12\\" title={[undefined]}>
                                      Once upon a time, Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin nec vulputate sapien. Fusce vel lacus fermentum, efficitur ipsum.
                                    </div>
                                  </Text>
                                </div>
                                <div className=\\"listing-controls\\">
                                  <Text className=\\"listing-due-date\\">
                                    <div className=\\"listing-due-date\\" title={[undefined]}>
                                      <Blueprint2.Icon className=\\"listing-due-icon\\" iconSize={12} icon=\\"time\\">
                                        <svg className=\\"pt-icon listing-due-icon\\" style={{...}} data-icon=\\"time\\" width={12} height={12} viewBox=\\"0 0 16 16\\">
                                          <title>
                                            time
                                          </title>
                                          <path d=\\"M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm1-6.41V4c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .28.11.53.29.71l2 2a1.003 1.003 0 0 0 1.42-1.42L9 7.59z\\" fillRule=\\"evenodd\\" />
                                        </svg>
                                      </Blueprint2.Icon>
                                      Due: 18/5, 13:24
                                    </div>
                                  </Text>
                                  <NavLink to=\\"/academy/sidequests/2/0\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
                                    <Route path=\\"/academy/sidequests/2/0\\" exact={[undefined]} strict={[undefined]} location={[undefined]}>
                                      <Link to=\\"/academy/sidequests/2/0\\" className={[undefined]} style={[undefined]} aria-current={false} replace={false}>
                                        <a className={[undefined]} style={[undefined]} aria-current={false} onClick={[Function]} href=\\"/academy/sidequests/2/0\\">
                                          <Blueprint2.Button className=\\"listing-skip-button\\" minimal={true} intent=\\"primary\\" icon=\\"flame\\">
                                            <button type=\\"button\\" className=\\"pt-button pt-minimal pt-intent-primary listing-skip-button\\" disabled={[undefined]} onClick={[undefined]} onKeyDown={[Function]} onKeyUp={[Function]}>
                                              <Blueprint2.Icon icon=\\"flame\\">
                                                <svg className=\\"pt-icon\\" style={{...}} data-icon=\\"flame\\" width={16} height={16} viewBox=\\"0 0 16 16\\">
                                                  <title>
                                                    flame
                                                  </title>
                                                  <path d=\\"M9.217 0c0 1.368.368 2.462 1.104 3.282C12.774 5.197 14 7.385 14 9.846c0 2.735-1.472 4.786-4.415 6.154 2.165-2.4 1.84-3.385-.368-6.4-2.342 1.2-1.967 2-1.592 3.6-.786 0-1.5 0-1.875-.4 0 .547.898 2 1.464 3.2-2.943-.82-6.092-5.744-4.988-6.154.736-.273 1.594-.137 2.575.41C3.575 5.333 5.047 1.915 9.217 0z\\" fillRule=\\"evenodd\\" />
                                                </svg>
                                              </Blueprint2.Icon>
                                              <span className=\\"pt-button-text\\">
                                                Skip Story &amp; Attempt
                                              </span>
                                              <Blueprint2.Icon icon={[undefined]} />
                                            </button>
                                          </Blueprint2.Button>
                                        </a>
                                      </Link>
                                    </Route>
                                  </NavLink>
                                </div>
                              </div>
                            </div>
                          </Blueprint2.Card>
                        </div>
                      </div>
                    </div>
                  </Blueprint2.Collapse>
                  <Blueprint2.Button onClick={[Function]} intent=\\"none\\" minimal={true} className=\\"collapse-button\\" icon=\\"caret-right\\">
                    <button type=\\"button\\" onClick={[Function]} className=\\"pt-button pt-minimal collapse-button\\" disabled={[undefined]} onKeyDown={[Function]} onKeyUp={[Function]}>
                      <Blueprint2.Icon icon=\\"caret-right\\">
                        <svg className=\\"pt-icon\\" style={{...}} data-icon=\\"caret-right\\" width={16} height={16} viewBox=\\"0 0 16 16\\">
                          <title>
                            caret-right
                          </title>
                          <path d=\\"M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 0 0 6 4.5v7a.495.495 0 0 0 .83.37l4-3.5c.1-.09.17-.22.17-.37z\\" fillRule=\\"evenodd\\" />
                        </svg>
                      </Blueprint2.Icon>
                      <span className=\\"pt-button-text\\">
                        Closed
                      </span>
                      <Blueprint2.Icon icon={[undefined]} />
                    </button>
                  </Blueprint2.Button>
                  <Blueprint2.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                    <div className=\\"pt-collapse\\" style={{...}}>
                      <div className=\\"pt-collapse-body\\" style={{...}} aria-hidden={false} />
                    </div>
                  </Blueprint2.Collapse>
                </div>
              </Blueprint2.Card>
            </div>
          </div>
        </ContentDisplay>
      </div>
    </Assessment>
  </Router>
</MemoryRouter>"
`;
